java/runtime/src/main/java/io/ray/runtime/util/SystemConfig.java

```diff
@@ -0,0 +1,27 @@
+package io.ray.runtime.util;
+
+import com.google.common.base.Preconditions;
+import com.google.gson.Gson;
+import java.util.HashMap;
+
+/** The utility class to read system config from native code. */
+public class SystemConfig {
+
+  private static Gson gson = new Gson();
+
+  /// A cache to avoid duplicated reading via JNI.
+  private static HashMap<String, Object> cachedConfigs = new HashMap<>(1024);
```

Comment:
Why do you give the hash map a init size?

Reply:
And can we use ConcurrentHashMap here instead of synchronizing `get` method?

Reply:
> Why do you give the hash map a init size?


Great catch. I introduced it to make sure it wouldn't rehash. But the config items size is not large, and no need to give it an initialized capacity.

Reply:
> And can we use ConcurrentHashMap here instead of synchronizing get method?

The reason is that if we don't sync the `get()` method, it'll be invoke native method twice, which is not a good practice in my view. I believe both are fine.